spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/AnnotatedServiceRegistrationBean.java

```diff
@@ -157,4 +178,37 @@ public AnnotatedServiceRegistrationBean setResponseConverters(
             ResponseConverterFunction... responseConverters) {
         return setResponseConverters(ImmutableList.copyOf(responseConverters));
     }
+
+    /**
+     * Returns example {@link HttpHeaders}.
+     */
+    @NotNull
+    public Collection<HttpHeaders> getExampleHeaders() {
+        return exampleHeaders;
+    }
+
+    /**
+     * Sets example {@link HttpHeaders}.
+     */
+    public AnnotatedServiceRegistrationBean setExampleHeaders(@NotNull Collection<HttpHeaders> exampleHeaders) {
+        this.exampleHeaders = exampleHeaders;
+        return this;
+    }
+
+    /**
+     * Returns sample requests of {@link #getService()}.
+     */
+    @NotNull
+    public Collection<ExampleRequest> getExampleRequests() {
+        return exampleRequests;
+    }
+
+    /**
+     * Sets sample requests for {@link #getService()}.
+     */
+    public AnnotatedServiceRegistrationBean setExampleRequests(
```

Comment:
How about moving this method and `getExampleRequests` up to `AbstractServiceRegistrationBean` so that we can remove duplicate code and use it for gRPC, Thrift and annotated services?
(we should override the method in `HttpServiceRegistrationBean` and throw the `UnsupportedOperationException` though.)

Reply:
Thank you for review. I fixed it in add07a2.

Reply:
@heowc I realized that each service needs different parameters for example requests:
- gRPC: service name, parameter name and the request object
  - `GrpcServiceRegistrationBean` can add multiple services using `GrpcServiceBuilder` so we need the service name to distinguish
- Annotated: parameter name and the request object
  - `AnnotatedServiceRegistrationBean` can add only one service so we don't need the service name.
- Thrift: request object
  - The service and method name can be inferred from the request object.

So we need different setters for each `ServiceRegistrationBean`, sorry about that. 😉 
Here's the thing we should do:
- Put the `ExampleRequest` back into `GrpcServiceRegistrationBean`
  - We are not going to use that class eventually but we need that now not to bring another breaking change.
  - Deprecate `ExampleRequest` class and related methods using `@Deprecated`.
    - This is because the class is only used in `GrpcServiceRegistrationBean`. It's not going to be used in other beans so it can confuse users.
  - Add new setter which will be: `setExampleRequests(String serviceType, String methodName, Object sampleRequest)`
- Add a setter to `AnnotatedServiceRegistrationBean`
  - We are not going to use `ExampleRequest` for here.
  - `setExampleRequests(String methodName, Object sampleRequest)`.
- Bring back the setters to `ThriftServiceRegistrationBean`.

If you have any questions, please let us know. 😄